### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Serchinastico**
### :tophat: What is the goal?

To update all dependencies to their latest version, including kotlin itself, gradle and coroutines.

### How is it being implemented?

* Upgrade dependencies: 
  * **gradle** `4.4` => `4.10.1`
  * **gradle plugin** `3.1.4` => `3.3.0`
  * **kotlin** `1.2.70` => `1.3.11`
  * **ktlint** `0.28.0` => `0.29.0`
  * **kotlinx-coroutines** `0.22.2` => `1.1.0`
  * **mockito-kotlin** `1.5.0` => `2.1.0`
  * **mockito-core** `2.16.0` => `2.23.4`
* Remove async/await dependency as it can be achieved with the coroutines library in the same way.
* Remove `AsyncExecutor`, coroutines now comes with a testing module to be able to specify where coroutines are going to be executed in a test.
* Run all tests in a `runBlocking` block to make them run sequentially.

### How can it be tested?

CI will tell 🤖

### CODE REVIEW

**Review by davideme — APPROVED** [review]

File: app/src/main/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKata.kt [hunk 1]
```diff
@@ -3,20 +3,24 @@ package com.karumi.kataloginlogoutkotlin
 import arrow.core.Either
 import arrow.core.left
 import arrow.core.right
+import kotlinx.coroutines.coroutineScope
```
Comment (tonilopezmr) [inline app/src/main/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKata.kt]: Do we want to add coroutines in this layer? or just add `suspend` keyword?

File: app/src/test/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKataTest.kt [hunk 2]
```diff
@@ -2,10 +2,11 @@ package com.karumi.kataloginlogoutkotlin
 
 import arrow.core.left
 import arrow.core.right
-import com.nhaarman.mockito_kotlin.whenever
+import com.nhaarman.mockitokotlin2.whenever
 import junit.framework.TestCase.assertEquals
-import junit.framework.TestCase.assertTrue
 import junit.framework.TestCase.assertFalse
+import junit.framework.TestCase.assertTrue
+import kotlinx.coroutines.runBlocking
```
Comment (tonilopezmr) [inline app/src/test/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKataTest.kt]: Why `LogInLogOutKata` uses `runBlocking` if there aren't coroutines in that file.

**Review by tonilopezmr — COMMENTED** [review]
I think we don't need `runBlocking` in `LogInLogOutKataTest.kt`

File: app/src/main/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKata.kt (hunk 1 above)
Comment (tonilopezmr) [reply app/src/main/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKata.kt]: **Nevermind** - you removed it

Comment (Serchinastico) [reply app/src/main/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKata.kt]: :)

File: app/src/test/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKataTest.kt (hunk 2 above)
Comment (Serchinastico) [reply app/src/test/java/com/karumi/kataloginlogoutkotlin/LogInLogOutKataTest.kt]: Right, I forgot about it when removing the async code from the use case, thanks toni

**Pull request merged**